File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.js
function solution(lottos, win_nums) {
  let [zero, match] = lottos.reduce(([zero, match], cur) => {
    if (cur==0) {
      return [zero+1, match];
    } else if (win_nums.includes(cur)) {
      return [zero, match+1];
    }
    return [zero, match];
  }, [0, 0]);

  let rank = [6, 6, 5, 4, 3, 2, 1];
  return [rank[zero+match], rank[match]];
  
}

File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    zeros = 0
    correct = 0
    for i in lottos:
        if i == 0:
            zeros += 1
        if i in win_nums:
            correct += 1
    maxcorrect = correct + zeros
    mincorrect = correct
    minrank = min([6, 7-mincorrect])
    maxrank = min([6, 7-maxcorrect])
    
    answer = [maxrank, minrank]
    return answer